docs/src/snforge-advanced-features/conditional-compilation.md

```diff
@@ -0,0 +1,80 @@
+# Conditional Compilation
+
+It is possible to build some contracts solely for testing purposes.
+This can be achieved by leveraging [Scarb features](../snforge-advanced-features/fork-testing.md).
+Configuration in `Scarb.toml` is done in the same manner as described in the Scarb documentation.
+Additionally, the `snforge test` command exposes the following flags: [--features](../appendix/snforge/test.md#-f---features-features),
+[--all-features](../appendix/snforge/test.md#--all-features) and [--no-default-features](../appendix/snforge/test.md#--no-default-features).
+
+## Contracts
+
+Firstly, define a contract in the `src` directory with a `#cfg(feature: (...))` attribute:
```

Comment:
Please put an emphasis that if one wants to declare a contract in a test it has to be in the package that is being tested, not in `tests`

Reply:
https://foundry-rs.github.io/starknet-foundry/testing/contracts-collection.html#how-contracts-are-collected

You can mention that it is like this bcs of the way snforge collects contracts and link this page

Reply:
Added a note about this